release-21.1: importccl: truncate row parsing errors

stevendanna opened the pull request:
Backport 1/1 commits from #73303.

/cc @cockroachdb/release

---

Errors encountered when processing rows are persisted to the jobs
table. If the error is too large, the error will be unable to mark
itself as reverting:

    pq: job 715036628973879297: could not mark as reverting: job-update:
    command is too large: 33561185 bytes (max: 4194304)

We now truncate the error to ensure we do not hit this limit.  We
could go further and completely remove the row content from the error
message.

Release note (bug fix): Error messages produced during import are now
truncated. Previously, import could potentially generate large error
messages that could not be persisted to the jobs table, resulting in a
failed import never entering the failed state and instead retrying
repeatedly.

Release justification: Bug fix for a bug that would result in an 
import running forever.

stevendanna committed:
importccl: truncate row parsing errors

Errors encountered when processing rows are persisted to the jobs
table. If the error is too large, the error will be unable to mark
itself as reverting:

    pq: job 715036628973879297: could not mark as reverting: job-update:
    command is too large: 33561185 bytes (max: 4194304)

We now truncate the error to ensure we do not hit this limit.  We
could go further and completely remove the row content from the error
message.

Release note (bug fix): Error messages produced during import are now
truncated. Previously, import could potentially generate large error
messages that could not be persisted to the jobs table, resulting in a
failed import never entering the failed state and instead retrying
repeatedly.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/73335)
<!-- Reviewable:end -->

stevendanna committed:
importccl,jobs: truncate all job system errors

In #73303 we truncated the row data in this error message to prevent
problems with large row data preventing the job status from being
saved.

However, truncating the row means that our
"experimental_save_rejected" option does not work as expected. This
feature previously saved entire rows and now it would have truncated
rows in some case.

Now, we only truncate the error when producing the error
string. Further, we also add truncation to the job system to prevent
other parts of the system from saving too much data to the job system.

Release note: None

stevendanna commented:
bors r=adityamaru

adityamaru commented:
Since this is a backport I'm canceling bors and just merging it.

bors r-
